test/tests/functional/pbs_sched_runjobwait.py

```diff
@@ -327,3 +327,73 @@ def test_throughput_ok(self):
 
         # Check that server received PBS_BATCH_AsynJob_ack request
         self.server.log_match("Type 97 request received", starttime=t)
+
+    def test_runhook_reject_comment_sched(self):
+        """
+        Test that when a runjob hook rejects a job, with runjob_wait
+        unset, that a job's comment is set correctly by sched
+        """
+        self.server.manager(MGR_CMD_SET, NODE,
+                            {"resources_available.ncpus": 4},
+                            id=self.mom.shortname)
+
+        self.server.manager(MGR_CMD_UNSET, SCHED, "job_run_wait")
+
+        jid1 = self.server.submit(Job())
+        self.server.expect(JOB, {"job_state": "R"}, id=jid1)
+        jid2 = self.server.submit(Job())
+        self.server.expect(JOB, {"job_state": "R"}, id=jid2)
+        jid3 = self.server.submit(Job())
+        self.server.expect(JOB, {"job_state": "R"}, id=jid3)
+
+        hook_txt = """
+import pbs
+
+e = pbs.event()
+j = e.job
+
+if not j.Resource_List["walltime"]:
+    e.reject("%s: no walltime specified" % (e.hook_name) )
+e.accept()"""
+        hk_attrs = {'event': 'runjob'}
+        self.server.create_import_hook('rj', hk_attrs, hook_txt)
+
+        t1 = time.time()
+        jid4 = self.server.submit(Job())
+        self.server.expect(JOB, {"job_state": "Q"}, id=jid4)
+        self.server.expect(JOB, "comment", op=SET, id=jid4)
+        ret = self.server.status(JOB, id=jid4)
+        jdata = ret[0]
+        comment = jdata["comment"]
+        self.assertIn("no walltime specified", comment)
```

Comment:
You can do this with an expect() with MATCH_RE

Reply:
I don't see a response to this comment.  I see no reason why you shouldn't use expect() here (and below)

Reply:
Oops, looks like I completely missed your comments on the test file. I'll make this change, thanks for the suggestion.